glotaran/parameter/test/test_parameter.py

```diff
@@ -1,169 +1,150 @@
+    assert parameters[0].minimum == -1
+    assert parameters[0].maximum == 1
+    assert not parameters[0].vary
 
-def test_parameter_options():
-    params = """
-    - ["5", 1, {non-negative: false, min: -1, max: 1, vary: false}]
-    - ["6", 4e2, {non-negative: true, min: -7e2, max: 8e2, vary: true}]
-    - ["7", 2e4]
-    """
-
-    params = load_parameters(params, format_name="yml_str")
-
-    assert params.get("5").value == 1.0
-    assert not params.get("5").non_negative
-    assert params.get("5").minimum == -1
-    assert params.get("5").maximum == 1
-    assert not params.get("5").vary
-
-    assert params.get("6").value == 4e2
-    assert params.get("6").non_negative
-    assert params.get("6").minimum == -7e2
-    assert params.get("6").maximum == 8e2
-    assert params.get("6").vary
-
-    assert params.get("7").value == 2e4
-    assert not params.get("7").non_negative
-    assert params.get("7").minimum == float("-inf")
-    assert params.get("7").maximum == float("inf")
-    assert params.get("7").vary
-
-
-def test_parameter_block_options():
-    params = """
-    block:
-        - 1.0
-        - [3.4, {vary: true}]
-        - {vary: false}
-    """
-
-    params = load_parameters(params, format_name="yml_str")
-    assert not params.get("block.1").vary
-    assert params.get("block.2").vary
-
-
-def test_parameter_set_from_group():
-    """Parameter extracted from group has correct values"""
-    group = load_parameters(
-        "foo:\n  - [\"1\", 123,{non-negative: true, min: 10, max: 8e2, vary: true, expr:'2'}]",
-        format_name="yml_str",
-    )
-    parameter = Parameter(full_label="foo.1")
-    parameter.set_from_group(group=group)
+    assert parameters[1].value == 4e2
+    assert parameters[1].non_negative
+    assert parameters[1].minimum == -7e2
+    assert parameters[1].maximum == 8e2
+    assert parameters[1].vary
 
-    assert parameter.value == 123
-    assert parameter.non_negative is True
-    assert np.allclose(parameter.minimum, 10)
-    assert np.allclose(parameter.maximum, 800)
-    assert parameter.vary is True
-    # Set to None since value and expr were provided?
-    assert parameter.expression is None
+    assert parameters[2].value == 2e4
+    assert not parameters[2].non_negative
+    assert parameters[2].minimum == float("-inf")
+    assert parameters[2].maximum == float("inf")
+    assert parameters[2].vary
 
 
 def test_parameter_value_not_numeric_error():
     """Error if value isn't numeric."""
-    with pytest.raises(TypeError, match="Parameter value must be numeric"):
-        Parameter(value="foo")  # type:ignore[arg-type]
+    with pytest.raises(TypeError):
+        Parameter(label="", value="foo")  # type:ignore[arg-type]
 
 
 def test_parameter_maximum_not_numeric_error():
     """Error if maximum isn't numeric."""
-    with pytest.raises(TypeError, match="Parameter maximum must be numeric"):
-        Parameter(maximum="foo")  # type:ignore[arg-type]
+    with pytest.raises(TypeError):
```

Comment:
Please add back the error message matching.

Reply:
not possible, since the raised errors are directly from attrs validators, which look like this:

```TypeError: ("'value' must be <class 'float'> (got 'foo' that is a <class 'str'>).", Attribute(name='value', default=nan, validator=_AndValidator(_validators=(<instance_of validator for type <class 'float'>>,)), repr=True, eq=True, eq_key=None, order=True, order_key=None, hash=None, init=True, metadata=mappingproxy({}), type='float', converter=<function Parameter.<lambda> at 0x7fb8e9e230a0>, kw_only=False, inherited=False, on_setattr=None), <class 'float'>, 'foo')
```

Please not the dynamic address.

Reply:
`match` uses a regex to match the error message partially so the memory address won't be a problem.
But since these are errors that face users they should also include the label of the parameter to be more helpful.
This could be done with a [custom validator](https://www.attrs.org/en/stable/examples.html#validators) and should be tested.
Moved to #1159